### PR TITLE
Feature/universal ea

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

**Comment by KuphJr** [comment]
How do I disable cache warming by default from within the adapter code?

File: packages/sources/universal/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
+# @chainlink/example-source-adapter
```
Comment (justinkaseman) [inline packages/sources/universal/CHANGELOG.md:1]: Clear me

File: packages/sources/universal/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@chainlink/universal-adapter",
+  "version": "0.2.17",
```
Comment (justinkaseman) [inline packages/sources/universal/package.json:3]:
```suggestion
  "version": "0.0.0",
```

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  const jobRunID = validator.validated.id
+
+  const httpQueries = validator.validated.data.httpQueries
+  const code = validator.validated.data.code
+  const secrets = validator.validated.data.secrets
+  const args = validator.validated.data.args
+
+  const data = { httpQueries, code, secrets, args }
+
+  // How do I create an new adapter-specfic config var instead of accessing process.env?
```
Comment (KuphJr) [inline packages/sources/universal/src/endpoint/sandbox.ts:63]: I don't quite remember how to add a new adapter-specific config var.

Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:63]: Do you mean reading it from `config`?

If so, add it in `makeConfig` of `config.ts`

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  const url = process.env['SANDBOX_URL']
+
+  if (!url)
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:66]: This logic would live better in `config.ts`

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  // TODO: inline-replace any instance of the string `$(secrets[x])` within the URL & headers w/ decrypted secrets
+
+  const options = { url, method: 'POST' as Method, data }
+
+  const sandboxResponse = await Requester.request<ResponseSchema>(options)
+
+  const adapterResponse = {
+    jobRunID,
+    statusCode: NaN,
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:81]: Why NaN?

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      jobRunID: request.id,
+      message: `set the URL of the FaaS sandbox using the SANDBOX_URL config variable`,
+      statusCode: 500,
+    })
+
+  // TODO: inline-replace any instance of the string `$(secrets[x])` within the URL & headers w/ decrypted secrets
+
+  const options = { url, method: 'POST' as Method, data }
+
+  const sandboxResponse = await Requester.request<ResponseSchema>(options)
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:77]: This request doesn't need to be auth'd?

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+  const adapterResponse = {
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:79]: Should follow existing patterns for handling responses (`Requester.success` and `Requester.withResult`). Might make this next bit of code cleaner.

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+import { Config, ExecuteWithConfig, InputParameters } from '@chainlink/ea-bootstrap'
+
+export const supportedEndpoints = ['sandbox']
+
+export interface ResponseSchema {
+  success?: hexstring
+  error?: string
+}
+
+export const description = 'This is an example endpoint description for example adapter.'
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:17]: Fill me

File: packages/sources/universal/src/endpoint/sandbox.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+export type TInputParameters = {
+  code: string
+  httpQueries?: HttpQuery[]
+  args?: string[]
+  secrets?: string
+}
+
+export const inputParameters: InputParameters<TInputParameters> = {
```
Comment (justinkaseman) [inline packages/sources/universal/src/endpoint/sandbox.ts:26]: Request struct for referencing:
```
 struct Request {
    Location codeLocation;
    Location secretsLocation;
    CodeLanguage language;
    string source; // Source code for Location.Inline or url for Location.Remote
    bytes secrets; // Encrypted secrets blob for Location.Inline or url for Location.Remote
    string[] args;
    HttpQuery[] queries;
  }
```

File: packages/tsconfig.test.json [hunk 10]
```diff
@@ -119,6 +119,9 @@
     {
       "path": "./scripts/tsconfig.test.json"
     },
+    {
+      "path": "./scripts/src/generate-readme/test/integration/readme-test-adapter/tsconfig.test.json"
```
Comment (justinkaseman) [inline packages/tsconfig.test.json:123]: Remove this. It's a bug in the generation script.

File: packages/tsconfig.json [hunk 11]
```diff
@@ -119,7 +119,9 @@
     {
       "path": "./scripts"
     },
-
+    {
```
Comment (justinkaseman) [inline packages/tsconfig.json:122]: Remove this. It's a bug in the generation script.

File: packages/sources/universal/src/endpoint/sandbox.ts (hunk 7 above)
Comment (KuphJr) [reply packages/sources/universal/src/endpoint/sandbox.ts:79]: Requester.withResult did not support returning an object containing a single `success` or `error` field, hence why I did not use it.  If you would prefer, I can make the required modification to the core framework to achieve this functionality.

File: packages/sources/universal/test/integration/fixtures.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+      'Connection',
+      'close',
+      'Vary',
+      'Accept-Encoding',
+      'Vary',
+      'Origin',
+    ])
+    // request with http requests, args & secrets
+    .post('/execute', {
+      httpRequests: [
```
[the remark this answers is not in the document]
Comment (KuphJr) [reply packages/sources/universal/test/integration/fixtures.ts:21]: Yes, I just have additional mocks that do not yet have integration tests which use the mocks.

File: packages/sources/universal/src/endpoint/sandbox.ts (hunk 7 above)
Comment (KuphJr) [reply packages/sources/universal/src/endpoint/sandbox.ts:121]: Resolved comment as discussed in DMs.  Requester.withResult does not support returning an object.

**Pull request closed without merging**